R/utils.R

```diff
@@ -144,7 +277,8 @@ check_null_or_scalar <- function(
   if (!is.null(x)) {
     if (length(x) > 1) {
       cli::cli_abort(
-        "{.arg {arg}} argument must be a scalar or {.val NULL}",
+        "{.arg {arg}} argument must be a scalar or {.val NULL},
+        not {.obj_type_friendly {x}}.",
         call = error_call
       )
     }
```

Comment:
check_null_or_scalar can be removed in favor of `check_string(x, allow_null = TRUE)`